engine/src/main/java/org/hibernate/validator/internal/engine/ValidationContext.java

```diff
@@ -68,6 +76,12 @@
 	private final Class<T> rootBeanClass;
 
 	/**
+	 * The metadata of the root bean.
+	 */
+	@Immutable
```

Comment:
Got it. In OGM we have `@EffectivelyImmutable` for a similar purpose (though it applies to initialization of the field during construction time, e.g. a post-construct callback, not lazy-init as done here). If needed we could add it to HV, too.

We don't need to handle those stereotypes with religious believe, it's just kind of an (open) experiment whether its helpful to us or not.